src/device/wg_controller.rs

```diff
@@ -252,11 +281,41 @@ async fn consolidate_wg_peers<
                 actual_peers.get(key),
                 requested_peers.get(key)
             );
+
             wireguard_interface
                 .add_peer(requested_peer.peer.clone())
                 .await?;
         }
 
+        // Check if we need to update sesion keeper
+        match (
+            session_keeper,
+            features.batching.is_some(),
+            requested_peer.batching_keepalive_interval,
+        ) {
+            (Some(sk), true, Some(interval)) => {
+                if sk.get_interval(key).await != Some(interval)
```

Comment:
Several small questions:
- Is SessionKeeper only a storage at this point which is not really used anywhere?
- It's a bit confusing as SessionKeeper receives a normal interval but here we compare it to the batching interval which means we compare two separate things :thinking: 
- `get_interval()` is not needed at all if this comparison gets removed IMO

It might just be that I missed some details here and there and it all makes sense though :D

Reply:
get_interval() is needed to compare against proxying peers that upgraded to direct ones by the means of checking their intervals(of course they might match and it simply means nothing needs to be done).